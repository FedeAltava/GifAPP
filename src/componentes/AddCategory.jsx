import { useState } from "react";

export const AddCategory = () => {

    const [inputValue, setInputValue] = useState('One Punch');

    const onImpuntChange = (event) => {
        console.log(event.target.value)
        setInputValue(event.target.value);
    }
    const onSubmit = (event)=>{
        event.preventDefault();
    }
    return (
        <form onSubmit ={(event)=>onSubmit(event)}>

            <input
                type="text"
                placeholder="Buscar Gif"
                value={inputValue}
                onChange={(event) => onImpuntChange(event)}
            />

        </form>


    )
}