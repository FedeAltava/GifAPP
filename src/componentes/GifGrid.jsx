import { GifItem } from "./GifItem";
import "../styles.css";
import { useFetchGif } from "../hooks/useFetchGifs";
export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGif(category);
  console.log(isLoading);
  return (
    <>
      <h1>-{category}:</h1>
      <div className="card-grid">
        {images.map((image) => {
          return <GifItem key={image.id} {...image} />;
        })}
      </div>
    </>
  );
};
