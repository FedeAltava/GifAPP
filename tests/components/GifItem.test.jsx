/* eslint-disable no-undef */

import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/componentes/GifItem"
describe('GiftItem component tests', () => {
    const title = 'Ramonet';
    const url = 'https://one-punch.com/saitama.jpg';

    it('should throw an error if propTypes are missing', () => {
        render( <GifItem />);
    });

    it('should match with snapshot', () => {
       const  {container} = render( <GifItem title ={title} url={url}/>);

        expect(container).toMatchSnapshot();
    });

    it('should render the image with URL and ALT', () => {
        render( <GifItem title ={title} url={url}/>);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
     });

     it('should render the title properly', () => {
        render( <GifItem title ={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();

     });
})