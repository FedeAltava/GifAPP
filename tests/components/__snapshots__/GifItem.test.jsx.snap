// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftItem component tests should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Ramonet"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Ramonet
    </p>
  </div>
</div>
`;
